File: VB/02_Text/SearchTextAndAddHyperlink.vb
Imports Spire.Pdf
Imports Spire.Pdf.Annotations
Imports Spire.Pdf.Texts

Namespace SearchTextAndAddHyperlink
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Specify the path to the input PDF file
            Dim input As String = "..\..\..\..\..\..\Data\SearchReplaceTemplate.pdf"

            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Load the PDF document from the specified input file
            doc.LoadFromFile(input)

            ' Get the first page of the document
            Dim page As PdfPageBase = doc.Pages(0)

            ' Create a PdfTextFinder object and specify the search options
            Dim finder As New PdfTextFinder(page)
            finder.Options.Parameter = Spire.Pdf.Texts.TextFindParameter.IgnoreCase

            ' Find all occurrences of the text "e-iceblue" on the page
            Dim finds As List(Of PdfTextFragment) = finder.Find("e-iceblue")

            ' Specify the URL for the hyperlink
            Dim url As String = "http://www.e-iceblue.com"

            ' Iterate over each found text fragment
            For Each find As PdfTextFragment In finds
                ' Create a PdfUriAnnotation object with the bounds of the found text
                Dim uri As New PdfUriAnnotation(find.Bounds(0))
                uri.Uri = url
                uri.Border = New PdfAnnotationBorder(1.0F)
                uri.Color = Color.Blue

                ' Add the annotation to the page's annotation widget collection
                page.Annotations.Add(uri)
            Next find

            ' Specify the output file name
            Dim result As String = "SearchTextAndAddHyperlink_out.pdf"

            ' Save the modified document to the specified output file
            doc.SaveToFile(result)

            ' Close the document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub

        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                System.Diagnostics.Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: CS/02_Text/SearchTextAndAddHyperlink.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Fields;
using Spire.Pdf.Annotations;
using Spire.Pdf.Graphics;
using Spire.Pdf.Actions;
using Spire.Pdf.General;
using Spire.Pdf.Texts;

namespace SearchTextAndAddHyperlink
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            String input = @"..\..\..\..\..\..\Data\SearchReplaceTemplate.pdf";
            PdfDocument doc = new PdfDocument();

            // Read a pdf file
            doc.LoadFromFile(input);

            // Get the first page of the pdf file
            PdfPageBase page = doc.Pages[0];

            // Create a PdfTextFinder using the first page
            PdfTextFinder finder = new PdfTextFinder(page);

            // Set the search parameter to ignore case
            finder.Options.Parameter = Spire.Pdf.Texts.TextFindParameter.IgnoreCase;

            // Find all occurrences of "e-iceblue" in the PDF and store them in a list
            List<PdfTextFragment> finds = finder.Find("e-iceblue");

            // Define the hyperlink URL
            String url = "http://www.e-iceblue.com";

            // Iterate through each found text fragment
            foreach (PdfTextFragment find in finds)
            {
                // Create a PdfUriAnnotation object to add a hyperlink for the searched text
                PdfUriAnnotation uri = new PdfUriAnnotation(find.Bounds[0]);
                uri.Uri = url;
                uri.Border = new PdfAnnotationBorder(1f);
                uri.Color = Color.Blue;

                // Add the annotation to the page's annotation widget
                page.Annotations.Add(uri);
            }

            String result = "SearchTextAndAddHyperlink_out.pdf";

            // Save the modified document to the output file path
            doc.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
